coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,359 @@
+        (0, 1, 2), default=0)
+
+
+@BlankLine.subaspect
+class BlankLineAfterProcedure:
+    """
+    Those found after procedures or functions.
+    """
+    class docs:
+        example = """
+        void proc(void){
+            printf("this does nothing");
+        }
+
+        int add(float a, float b){
+            return a + b;
+        }
+        """
+        example_language = 'C'
+        importance_reason = """
+        Having a specific and reasonable number of blank lines after every
+        procedures improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `BlankLintAfterProcedure` issues can be fixed specifying (and of
+        course using) a reasonable number of blank lines to use after
+        procedures' definition.
+        """
+    blank_lines_after_procedures = Taste[int](
+        'Represents the number of blank lines to use after a procedure or a'
+        ' function', (0, 1, 2), default=1)
+
+
+@BlankLine.subaspect
+class BlankLineAfterClass:
+    """
+    Those found after classes' definitions.
+    """
+    class docs:
+        example = """
+        class SomeClass:
+            def __init__(self):
+                raise RuntimeError('Never instantiate this class')
+
+
+        def func():
+            pass
+        """
+        example_language = 'Python 3'
+        importance_reason = """
+        Having a specific number of blank lines after every classes'
+        definitions declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class'
+        'definition.', (1, 2), default=2)
+
+
+@BlankLine.subaspect
+class NewlineAtEOF:
+    """
+    Newline character (usually '\\\\n', aka CR) found at the end of file.
+    """
+    class docs:
+        example = """
+        def do_nothing():
+            pass
+        """ + ('\n')
+        example_language = 'Python 3'
+        importance_reason = """
+        A text file consists of a series of lines, each of which ends with a
+        newline character (\\\\n). A file that is not empty and does not end
+        with a newline is therefore not a text file.
+
+        It's not just bad style, it can lead to unexpected behavior, utilities
+        that are supposed to operate on text files may not cope well with files
+        that don't end with a newline.
+        """
+        fix_suggestions = """
+        `NewlineAtEOF` issues can be fixed by simply adding a newline at the
+        end of the file.
+        """
+    newline_at_EOF = Taste[bool](
+        'If ``True``, enforce a newline at End Of File.',
+        (True, False), default=True)
+
+
+@Spacing.subaspect
+class SpacesAroundOperator:
+    """
+    Spacing around operators.
+    """
+    class docs:
+        example = """
+        def f(a, x, y):
+            return 37++a[42-x :  y**3]
```

Comment:
This is supposed to be demonstrating whitespaces around operators. It is still gobbly-gook. You dont have a functioning piece of code with one obvious problem that the reader focuses on.
Now you have changed it, someone on Python 2 could try to find a,x,y that are not an IndexError, but on Python 3 it is always a TypeError because of the float division.